packager/certfetcher/certfetcher.go

```diff
@@ -119,15 +119,16 @@ func New(email string, eabKid string, eabHmac string, certSignRequest *x509.Cert
 		}
 	}
 
-	// Theoretically, this should always be set to false as users should have pre-registered for access
-	// to the ACME CA and agreed to the TOS.
-	// TODO(banaag): revisit this when trying the class out with Digicert CA.
-	if !shouldRegister {
-		acmeUser.Registration = new(registration.Resource)
+	var reg *registration.Resource
+	// Check if we already have an account.
+	if reg, err = client.Registration.ResolveAccountByKey(); err == nil {
+		acmeUser.Registration = reg
 	} else {
-		var reg *registration.Resource
-		var err error
-
+		// We need to reset the LEGO client.
```

Comment:
Is it feasible to add unit tests for this change? e.g. To prove that it's necessary to reset the client.

Reply:
Unfortunately, no.  I tested it against real CAs and it is necessary.

Reply:
OK. Add "after calling Registration.ResolveAccountByKey()" to the comment (if that's right).

Reply:
Done.